internal/runner/discover.go

```diff
@@ -0,0 +1,48 @@
+package runner
+
+import (
+	"fmt"
+	"io/fs"
+
+	"github.com/DrJosh9000/zzglob"
+)
+
+type DiscoveryPattern struct {
+	IncludePattern string
+	ExcludePattern string
+}
+
+func discoverTestFiles(pattern DiscoveryPattern) ([]string, error) {
+	parsedPattern, err := zzglob.Parse(pattern.IncludePattern)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing pattern %q", pattern)
+	}
+
+	parsedExcludePattern, err := zzglob.Parse(pattern.ExcludePattern)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing exclude pattern %q", pattern.ExcludePattern)
+	}
+
+	discoveredFiles := []string{}
+
+	// Use the Glob function to traverse the directory recursively
+	// and append the matched file paths to the discoveredFiles slice
+	parsedPattern.Glob(func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			fmt.Printf("Error walking: %v\n", err)
+		}
+		if d.IsDir() {
```

Comment:
non-blocking question: what this does? is this one checking the directory doesn't have test files?

Reply:
`Glob` works similarly to a filepath walk, which calls the callback both with matching files _and_ the intermediate directories. This provides a means to skip a particular directory for whatever reason.